src/core/server/saved_objects/service/lib/kql_utils.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { fromKueryExpression, toElasticsearchQuery } from '@kbn/es-query';
+import { SavedObjectsIndexPattern } from './cache_index_patterns';
+
+export const convertKqlToElasticSearchQuery = (
```

Comment:
IMHO this would make sense inside https://github.com/elastic/kibana/blob/9885200c3e1d977169c4310393716c857b6bc817/packages/kbn-es-query/src/kuery/ast/ast.js